cmake/Install.cmake

```diff
@@ -9,4 +9,107 @@
+                set(ly_install_directory_DESTINATION ${CMAKE_CURRENT_LIST_DIR})
+            else()
+                cmake_path(ABSOLUTE_PATH ly_install_directory_DESTINATION BASE_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})
+            endif()
+            # take out the last directory since install asks for the destination of the folder, without including the fodler itself
+            cmake_path(GET ly_install_directory_DESTINATION PARENT_PATH ly_install_directory_DESTINATION)
+            cmake_path(RELATIVE_PATH ly_install_directory_DESTINATION BASE_DIRECTORY ${LY_ROOT_FOLDER})       
+        endif()
+
+        unset(exclude_patterns)
+        if(ly_install_directory_EXCLUDE_PATTERNS)
+            foreach(exclude_pattern ${ly_install_directory_EXCLUDE_PATTERNS})
+                list(APPEND exclude_patterns PATTERN ${exclude_pattern} EXCLUDE)
+            endforeach()
+        endif()
+        
+        # Exclude cmake since that has to be generated
+        list(APPEND exclude_patterns PATTERN CMakeLists.txt EXCLUDE)
+        list(APPEND exclude_patterns PATTERN *.cmake EXCLUDE)
+
+        # Exclude python-related things that dont need to be installed
+        list(APPEND exclude_patterns PATTERN __pycache__ EXCLUDE)
+        list(APPEND exclude_patterns PATTERN *.egg-info EXCLUDE)     
+
+        install(DIRECTORY ${directory}
+            DESTINATION ${ly_install_directory_DESTINATION}
+            COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME} # use the deafult for the time being
+            ${exclude_patterns}
+        )
+    endforeach()
+
+endfunction()
+
+#! ly_install_files: specifies files to be copied to the install layout at install time
+#
+# \arg:FILES files to install
+# \arg:DESTINATION (optional) destination to install the directory to (relative to CMAKE_PREFIX_PATH)
+# \arg:PROGRAMS (optional) indicates if the files are programs that should be installed with EXECUTE permissions
+#
+# \notes: refer to cmake's install(FILES/PROGRAMS documentation for more information
+#
+function(ly_install_files)
+
+    set(options PROGRAMS)
+    set(oneValueArgs DESTINATION)
+    set(multiValueArgs FILES)
+
+    cmake_parse_arguments(ly_install_files "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if(NOT ly_install_files_FILES)
+        message(FATAL_ERROR "You must provide a list of files to install")
+    endif()
+    if(NOT ly_install_files_DESTINATION)
+        message(FATAL_ERROR "You must provide a destination to install filest to")
+    endif()
+
+    unset(files)
+    foreach(file ${ly_install_files_FILES})
+        cmake_path(ABSOLUTE_PATH file)
+        list(APPEND files ${file})
+    endforeach()
+
+    if(ly_install_files_PROGRAMS)
+        set(install_type PROGRAMS)
+    else()
+        set(install_type FILES)
+    endif()
+
+    install(${install_type} ${files}
+        DESTINATION ${ly_install_files_DESTINATION}
+        COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME} # use the deafult for the time being
```

Comment:
typo: deafult
```suggestion
        COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME} # use the default for the time being
```

Reply:
Also shouldn't we make the COMPONENT_NAME an optional parameter that defaults to `CMAKE_INSTALL_DEFAULT_COMPONENT_NAME`?

Reply:
didnt want to expose components yet since I think most component division will be based on type of things and not targets
If we need to expose components to the targets, I will add the parameter with a default